lib/git_repository.rb

```diff
@@ -91,6 +91,21 @@ def self.directory(path)
     path.split("/")[0..-2].join("/")
   end
 
+  def deepest_existing_dir(path, commit_oid = nil)
+    path ||= '/'
+    dirs = path.split('/')
+
+    dir = nil
+    Array(0..dirs.length - 1).reverse.each do |i|
+      if dir.nil?
+        path = dirs[0..i].join('/')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.